file-formats/aifi/type/zif_aff_aifi_v1.intf.abap

```diff
@@ -0,0 +1,523 @@
+INTERFACE zif_aff_aifi_v1
+  PUBLIC.
+
+  "! $values { @link zif_aff_aifi_v1.data:co_selection_type }
+  "! $default { @link zif_aff_aifi_v1.data:co_selection_type.single }
+  TYPES ty_selection_type TYPE c LENGTH 1.
+  CONSTANTS:
+    BEGIN OF co_selection_type,
+      "! <p class="shorttext">Single Selection</p>
+      "! Single selection
+      single   TYPE ty_selection_type VALUE ' ',
+      "! <p class="shorttext">Multiple Selection</p>
+      "! Multiple selection
+      multiple TYPE ty_selection_type VALUE 'M',
+      "! <p class="shorttext">Document ID Selection</p>
+      "! Document ID selection
+      document TYPE ty_selection_type VALUE 'D',
+    END OF co_selection_type.
+
+  "! $values { @link zif_aff_aifi_v1.data:co_sap_raw_structure_type }
+  "! $default { @link zif_aff_aifi_v1.data:co_sap_raw_structure_type.raw }
+  TYPES ty_sap_raw_structure_type TYPE c LENGTH 1.
+  CONSTANTS:
+    BEGIN OF co_sap_raw_structure_type,
+      "! <p class="shorttext">Source Structure (raw for inbound, SAP for outbound)</p>
+      "! Source structure (raw for inbound, SAP for outbound)
+      raw TYPE ty_sap_raw_structure_type VALUE ' ',
+      "! <p class="shorttext">Destination Structure (SAP for inbound, raw for outbound)</p>
+      "! Destination structure (SAP for inbound, raw for outbound)
+      sap TYPE ty_sap_raw_structure_type VALUE 'S',
```

Comment:
We strongly recommend having name of component and title in sync.

Reply:
The title is what appears in the drop down. How would I make a component name from that? I think that´s not prudent here.

Reply:
Maybe I don't understand this field correctly.
The field has currently two possible values: `Source Structure (raw for inbound, SAP for outbound)` and `Destination Structure (SAP for inbound, raw for outbound)`. Therefore I understood that raw is the technical value for Source Structure and sap for Destination structure. But checking your title again, it seems to be different?!

Reply:
RAW and SAP structure can be source or destination, depending on the interface direction. Only thing I can imagine to simply this: Change Titles to Source Structure and Destination Structure. Leave away the hint in brackets. And then rename the components to source_structure and destination_structure . Would that be acceptable and comprehensible to you? I think can move the complicated detail here to the documentation.

Reply:
For me, it would be less confusing. But I understood it, now. As expert on your domain, I would like to leave this decision up to you.

Reply:
hm. in fact never felt comfortable with that state and adjusted it. pls double check from your end.

Reply:
Thanks, looks good to me!